### PR TITLE
Use HTTPStatus constants in place of literals in tests

### PR DESCRIPTION
**Pull request by dklimpel**
### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

Signed-off-by: Dirk Klimpel dirk@klimpel.org

### CODE REVIEW

File: changelog.d/13297.misc [hunk 1]
```diff
@@ -0,0 +1 @@
+Use HTTPStatus constants in place of literals in tests.
```
Comment (clokep) [inline changelog.d/13297.misc:1]:
```suggestion
Use `HTTPStatus` constants in place of literals in tests.
```

**Review by clokep — APPROVED** [review]
LGTM, thanks!

**Comment by richvdh** [comment]
I'll not insist on this being reverted, but per #13298: I'm not entirely convinced this is a direction we should be going in, and I'd like us to discuss it before propagating the change more widely.

**Comment by clokep** [comment]
> I'm not entirely convinced this is a direction we should be going in, and I'd like us to discuss it before propagating the change more widely.

I was under the impression we had started moving towards these for new tests and this was just a consistency thing. I don't have strong feelings either way.

**Pull request merged**